packages/core/src/delegation/DelegationDecoder.ts

```diff
@@ -77,7 +77,7 @@ export type DelegationNodeRecord = Omit<IDelegationNode, 'id'>
 
 export type DelegationNodeId = Hash
 
-export function decodeDelegationNode(
+export function delegationNodeFromChain(
   encoded: Option<DelegationDelegationHierarchyDelegationNode>
 ): DelegationNodeRecord | null {
```

Comment:
are we not just throwing now when something does not exist? returning null here seems inconsistent